### ISSUE TITLE
GQL queries origin

### ISSUE THREAD

**Issue by joshua-seals** [dbeaver/cloudbeaver#2934]
Hello, first thanks in advance for the help and also really enjoying learning about cloudbeaver. Very excited to replace pgadmin with it! 

Are the GQL queries server-side or basically being issued from the browser? My setup is in K8S and ideally gql would always query 0.0.0.0/api/ but it seems to pickup the hostname of the external connection string despite server.serverHost: "0.0.0.0" setting.

**Comment by LonwoLonwo** [comment]
Hello @joshua-seals 

Thanks for your interest in CloudBeaver!

Please clarify your question one more time. And add more examples. Or, maybe a video or screens.
Your browser sends GQL queries, and the server works with them.

**Comment by joshua-seals** [comment]
Ok that seems to be the answer to my question, from browser to gql endpoint. 

I have a reverseProxy setup which uses a /private scheme in resty. So a user would deploy an app (cloudbeaver in this case), then that user and their session gets matched up to the newly provisioned pod/app in k8s and ambassador authorizes the request, sets headers for the service and maps the user to the new deployment. 

Basically the gql queries are causing error as they are attempting to hit the endpoint /private/user/rand-string/ endpoint which get's blocked by nginx. I also see all the websocket trafic being blocked so just needed some clarity

**Comment by LonwoLonwo** [comment]
Sorry for the late response.

> Basically the gql queries are causing error

Could you please provide some steps for reproduction? 

> I also see all the websocket trafic being blocked so just needed some clarity

How exactly is it blocked?

Also, for now, we think that your Nginx can be configured incorrectly.

**Comment by mr-anton-t** [comment]
@joshua-seals, please check your NGINX configuration  and check the response code for the next request:

```bash
  curl  -i -N -k \
  --http1.1 \
  'https://****server_address***/api/ws' \
  -H 'Upgrade: websocket' \
  -H 'Origin: https://****server_address***' \
  -H 'Cookie: cb-session-id=ehl2n**example**f1rl1bk15' \
  -H 'Connection: close' \
  -H 'Sec-WebSocket-Key: 0VpP**********vb4w==' \
  -H 'Sec-WebSocket-Version: 13' \
  -H 'Sec-WebSocket-Extensions: permessage-deflate; client_max_window_bits' 
```

(*) When making a request, please specify the correct Cookie,  Sec-WebSocket-Key, and Server address

The following server response code should be returned - `101 Switching Protocols`

**Comment by LonwoLonwo** [comment]
It's been a while since there has been an update here. Please let me know if the issue is still present and provide additional information.
